syntax = "proto3";
package motion_server;

message SetMotionRequest {
  string name = 1;
  optional int32 priority = 2;
  optional bool repeat = 3;
  optional bool clear = 4;
}

message SetMotionReply {
  bool success = 1;
}

message ClearMotionRequest {
  optional int32 priority = 1;
}

message ClearMotionReply {
  bool success = 1;
}

message StopRepeatRequest {
  optional int32 priority = 1;
}

message StopRepeatReply {
  bool success = 1;
}

message SetWaitRequest {
  float time = 1;
  optional int32 priority = 2;
  optional bool clear = 3;
}

message SetWaitReply {
  bool success = 1;
}

message SetPosRequest {
  optional float pan = 1;
  optional float tilt = 2;
  optional int32 priority = 3;
}

message SetPosReply {
  bool success = 1;
}

message SetVelRequest {
  optional float pan = 1;
  optional float tilt = 2;
  optional int32 priority = 3;
}

message SetVelReply {
  bool success = 1;
}

message SetAccRequest {
  optional float pan = 1;
  optional float tilt = 2;
  optional int32 priority = 3;
}

message SetAccReply {
  bool success = 1;
}

service MotionServerService {
    rpc SetMotion(SetMotionRequest)
        returns (SetMotionReply){}
    rpc ClearMotion(ClearMotionRequest)
        returns (ClearMotionReply){}
    rpc StopRepeat(StopRepeatRequest)
        returns (StopRepeatReply){}
    rpc SetWait(SetWaitRequest)
        returns (SetWaitReply){}
    rpc SetPos(SetPosRequest)
        returns (SetPosReply){}
    rpc SetVel(SetVelRequest)
        returns (SetVelReply){}
    rpc SetAcc(SetAccRequest)
        returns (SetAccReply){}
}
